cni/network/invoker_azure.go

```diff
@@ -92,13 +92,18 @@ func (invoker *AzureIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, er
 			nwCfg6.IPAM.Subnet = invoker.nwInfo.Subnets[1].Prefix.String()
 		}
 
-		addResult.ipv6Result, err = invoker.plugin.DelegateAdd(nwCfg6.IPAM.Type, &nwCfg6)
+		var ipv6Result *cniTypesCurr.Result
+		ipv6Result, err = invoker.plugin.DelegateAdd(nwCfg6.IPAM.Type, &nwCfg6)
 		if err != nil {
 			err = invoker.plugin.Errorf("Failed to allocate v6 pool: %v", err)
+		} else {
+			result.IPs = append(result.IPs, ipv6Result.IPs...)
+			result.Routes = append(result.Routes, ipv6Result.Routes...)
+			addResult.ipv6Enabled = true
```

Comment:
Shouldn't we first check if Ipv6 result is empty or not before setting addResult.ipv6Enabled = true ?